includes/gutenberg/shortcodes/shortcodes-to-blocks.php

```diff
@@ -0,0 +1,421 @@
+// Render navigation to a post form
+//
+function buddyforms_block_navigation( $attributes ) {
+	global $buddyforms;
+
+	if ( isset( $attributes['bf_form_slug'] ) && isset( $buddyforms[ $attributes['bf_form_slug'] ] ) ) {
+
+		$args = array(
+			'form_slug'  => $attributes['bf_form_slug'],
+			'separator'  => empty( $attributes['bf_nav_separator'] ) ? '' : $attributes['bf_nav_separator'],
+			'label_add'  => empty( $attributes['bf_label_add'] ) ? 'Add' : $attributes['bf_label_add'],
+			'label_view' => empty( $attributes['bf_label_view'] ) ? 'View' : $attributes['bf_label_view'],
+		);
+
+		if ( isset( $attributes['bf_nav_style'] ) ) {
+			switch ( $attributes['bf_nav_style'] ) {
+				case 'buddyforms_button_view_posts':
+					return buddyforms_button_view_posts( $args );
+					break;
+				case 'buddyforms_button_add_new':
+					return buddyforms_button_add_new( $args );
+					break;
+
+			}
+		}
+
+
+		return buddyforms_nav( $args );
+
+
+	} else {
+		return '<p>' . __( 'Please select a form in the block settings sidebar!', 'buddyforms' ) . '</p>';
+	}
+}
+
+
+//
+// Render a form submissions
+//
+function buddyforms_block_list_submissions( $attributes ) {
+	global $buddyforms;
+
+//	print_r($attributes);
+	$display = false;
+
+	if ( ! is_user_logged_in() && $attributes['bf_rights'] == 'public' ) {
+		$display = true;
+	}
+	if ( is_user_logged_in() && $attributes['bf_rights'] == 'public' ) {
+		$display = true;
+	}
+	if ( is_user_logged_in() && $attributes['bf_rights'] == 'private' ) {
+		$display = true;
+	}
+	$user = wp_get_current_user();
+	if ( is_user_logged_in() && in_array( $attributes['bf_rights'], (array) $user->roles ) ) {
+		$display = true;
+	}
+
+	if ( $display ) {
+		if ( isset( $attributes['bf_form_slug'] ) && isset( $buddyforms[ $attributes['bf_form_slug'] ] ) ) {
+
+			$list_style           = empty( $attributes['bf_list_posts_style'] ) ? 'list' : $attributes['bf_list_posts_style'];
+			$posts_per_page       = empty( $attributes['bf_posts_per_page'] ) ? '10' : $attributes['bf_posts_per_page'];
+			$filter_by_author     = empty( $attributes['bf_by_author'] ) ? 'logged_in_user' : $attributes['bf_by_author'];
+			$filter_by_author_ids = empty( $attributes['bf_author_ids'] ) ? '' : $attributes['bf_author_ids'];
+			$bf_by_form           = empty( $attributes['bf_by_form'] ) ? 'form' : $attributes['bf_by_form'];
+
+			ob_start();
+			buddyforms_blocks_the_loop(
+				array(
+					'form_slug'        => $attributes['bf_form_slug'],
+					'list_posts_style' => $list_style,
+					'posts_per_page'   => $posts_per_page,
+					'query_option'     => $filter_by_author,
+					'author_ids'       => $filter_by_author_ids,
+					'bf_by_form'       => $bf_by_form
+				)
+			);
+			$tmp = ob_get_clean();
+
+			return $tmp;
+
+		} else {
+			return '<p>' . __( 'Please select a form in the block settings sidebar!', 'buddyforms' ) . '</p>';
+		}
+	}
+
+}
+
+//
+// Query posts for the submission lists
+//
+function buddyforms_blocks_the_loop( $args ) {
```

Comment:
@svenl77 are this function exist in another file?

Reply:
no, it is similar to the shortcode buddyforms_the_loop.
I started use this function but during development, I decided to overwrite it and create a new function for the block. 

We have to many dependencies in the shortcode function and I not wanted to break other code